pkg/process/checks/interval.go

```diff
@@ -0,0 +1,101 @@
+
+	// The interval, in seconds, at which we will run each check. If you want consistent
+	// behavior between real-time you may set the Container/ProcessRT intervals to 10.
+	// Defaults to 10s for normal checks and 2s for others.
+	configProcessInterval     = configIntervals + ".process"
+	configRTProcessInterval   = configIntervals + ".process_realtime"
+	configContainerInterval   = configIntervals + ".container"
+	configRTContainerInterval = configIntervals + ".container_realtime"
+	configConnectionsInterval = configIntervals + ".connections"
+)
+
+var (
+	defaultIntervals = map[string]time.Duration{
+		ProcessCheckName:       ProcessCheckDefaultInterval,
+		RTProcessCheckName:     RTProcessCheckDefaultInterval,
+		ContainerCheckName:     ContainerCheckDefaultInterval,
+		RTContainerCheckName:   RTContainerCheckDefaultInterval,
+		ConnectionsCheckName:   ConnectionsCheckDefaultInterval,
+		PodCheckName:           PodCheckDefaultInterval,
+		DiscoveryCheckName:     ProcessDiscoveryCheckDefaultInterval,
+		ProcessEventsCheckName: config.DefaultProcessEventsCheckInterval,
+	}
+
+	configKeys = map[string]string{
+		ProcessCheckName:     configProcessInterval,
+		RTProcessCheckName:   configRTProcessInterval,
+		ContainerCheckName:   configContainerInterval,
+		RTContainerCheckName: configRTContainerInterval,
+		ConnectionsCheckName: configConnectionsInterval,
+	}
+)
+
+// GetDefaultInterval returns the default check interval value
+func GetDefaultInterval(checkName string) time.Duration {
+	return defaultIntervals[checkName]
+}
+
+// GetInterval returns the configured check interval value
+func GetInterval(checkName string) time.Duration {
+
+	switch checkName {
+	case DiscoveryCheckName:
+		// We don't need to check if the key exists since we already bound it to a default in InitConfig.
+		// We use a minimum of 10 minutes for this value.
+		discoveryInterval := config.Datadog.GetDuration("process_config.process_discovery.interval")
+		if discoveryInterval < discoveryMinInterval {
+			discoveryInterval = discoveryMinInterval
+			_ = log.Warnf("Invalid interval for process discovery (<= %s) using default value of %[1]s", discoveryMinInterval.String())
+		}
+		return discoveryInterval
+
+	case ProcessEventsCheckName:
+		eventsInterval := config.Datadog.GetDuration("process_config.event_collection.interval")
+		if eventsInterval < config.DefaultProcessEventsMinCheckInterval {
+			eventsInterval = config.DefaultProcessEventsCheckInterval
+			_ = log.Warnf("Invalid interval for process_events check (< %s) using default value of %s",
+				config.DefaultProcessEventsMinCheckInterval.String(), config.DefaultProcessEventsCheckInterval.String())
+		}
+		return eventsInterval
+
+	default:
+		defaultInterval := defaultIntervals[checkName]
+		configKey, ok := configKeys[checkName]
+		if !ok || !config.Datadog.IsSet(configKey) {
+			_ = log.Errorf("missing check interval for '%s', you must set a default", checkName)
```

Comment:
It might not be a good idea to log an error here. AFAIK most customers don't override the default check intervals, and I don't necessarily think we should be telling them they need to.

Nit: Also, I think the convention is to not explicitly ignore the error for logs, do you mind removing the `_ =`. In the past, people have been confused when I added this and it would be nice to keep a consistent style.

Reply:
Point taken, this was moved rather mechanically though, so the logging part exists today.

Reply:
Oh interesting. I didn't notice that. I think it might be worthwhile removing while we're here.

Reply:
I agree with you, it was probably done this way when we weren't binding defaults to these in `pkg/config` so removing the log makes sense!